### PR TITLE
search page 

### PR DESCRIPTION
**Pull request by Mrudul111**
Implemented search page when we click on the tab with flag and country code

### CODE REVIEW

**Comment by gopavasanth** [comment]
Hi @Mrudul111 , Thanks for your quick PR! I appreciate your efforts :)

Please change the app bar color on the search page, and change the search string placeholder to "Search by country name"? And make a dropdown in the home screen beside the country code.
| a | b |
| - | - |
| image |  Screenshot_1696351707 |

**Comment by Mrudul111** [comment]
Sure thing

File: lib/src/search_page.dart [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+      body: Column(
+        children: [
+          CountrySearchBar(
+            allCountries: allCountries,
+            onSearch: updateFilteredCountries,
+          ),
+          Expanded(
+            child: ListView.builder(
+              itemCount: filteredCountries.length,
+              itemBuilder: (context, index) {
```
Comment (gopavasanth) [inline lib/src/search_page.dart:45]: You can probably add dividers to each item in the list so to easily read the values.

File: lib/src/search_page.dart [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+                final country = filteredCountries[index];
+                return GestureDetector(
+                  onTap: () {
+                    // Pass the selected country back to the previous page
+                    Navigator.pop(context, country);
+                  },
+                  child: Container(
+                    padding: EdgeInsets.symmetric(vertical: 10, horizontal: 16),
+                    child: Row(
```
Comment (gopavasanth) [inline lib/src/search_page.dart:54]: You could give horizontal padding to the row as they feel they are a bit to the extreme.

**Comment by Mrudul111** [comment]
I have made the appear color white and font black for the contrast
also added the dropdown there and sized it accordingly.

File: lib/src/search_page.dart [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+  _CountrySearchBarState createState() => _CountrySearchBarState();
+}
+
+class _CountrySearchBarState extends State<CountrySearchBar> {
+  TextEditingController _searchController = TextEditingController();
+
+  @override
+  void initState() {
+    super.initState();
+    _searchController.addListener(_onSearchTextChanged);
```
Comment (gopavasanth) [inline lib/src/search_page.dart:107]: dispose of the controller before widget dispose as to remove the listener

File: lib/main.dart [hunk 4]
```diff
@@ -4,6 +4,7 @@ import 'package:whatsapp_karo/src/home.dart';
 
 void main() {
   runApp(MaterialApp(
+    debugShowCheckedModeBanner: false,
```
Comment (gopavasanth) [inline lib/main.dart:7]: No need for this. This helps us know it's in debug more we are running.

**Review by gopavasanth — CHANGES_REQUESTED** [review]

File: lib/src/search_page.dart (hunk 2 above)
Comment (Mrudul111) [reply lib/src/search_page.dart:54]: 20 would be fine?

Comment (gopavasanth) [reply lib/src/search_page.dart:54]: Anything would be fine, just it should look good

File: lib/src/search_page.dart (hunk 1 above)
Comment (Mrudul111) [reply lib/src/search_page.dart:45]: how is the spacing and divider

File: lib/src/search_page.dart (hunk 3 above)
Comment (Mrudul111) [reply lib/src/search_page.dart:107]: done

File: lib/src/search_page.dart (hunk 1 above)
Comment (gopavasanth) [reply lib/src/search_page.dart:44]: Screenshot 2023-10-03 at 11 50 53 PM

Make sure these don't overflow

Comment (gopavasanth) [reply lib/src/search_page.dart:44]: Screenshot 2023-10-03 at 11 56 43 PM

Also, handle them properly when font size is increased (display settings in mobile)

Also, make the divider color a bit light, it feels too dark

Comment (Mrudul111) [reply lib/src/search_page.dart:44]: sure on it

Comment (immadisairaj) [reply lib/src/search_page.dart:44]: Can you add some screenshots @Mrudul111? I am currently outside and will only be able to look at this in the night.

Comment (Mrudul111) [reply lib/src/search_page.dart:44]: sure

Comment (Mrudul111) [reply lib/src/search_page.dart:44]: image

Comment (Mrudul111) [reply lib/src/search_page.dart:44]: image

Comment (Mrudul111) [reply lib/src/search_page.dart:44]: sorry for the delay man but all the issues are fixed now

**Comment by immadisairaj** [comment]
I am really sorry @Mrudul111, I was trying to add another commit to your branch and messed up. Can you push the local commits you have to open this PR again?

Really sorry again.

**Comment by Mrudul111** [comment]
new pr made

**Comment by immadisairaj** [comment]
> new pr made

Thank you and sorry. :)

**Pull request closed without merging**